### ISSUE TITLE
Different resolvers than SBT

### ISSUE THREAD

**Issue by jkutner** [typesafehub/activator#1057]
Activator uses different resolvers than SBT. This is a major problem for CI, PaaS, and other cloud based services that use SBT to build applications created and developed with Activator.

The issue can easily be reproduced with this very small project:
https://github.com/jkutner/activator-sbt-bug

The project contains a single dependency:

```scala
"com.typesafe.sbtrc" % "actor-client-2-11" % "0.3.1"
```

When sbt is run, the dependency cannot be resolved:

```sh-session
$ rm -rf ~/.ivy2/cache/com.typesafe.sbtrc/
$ sbt update
...
[warn] 	module not found: com.typesafe.sbtrc#actor-client-2-11;0.3.1
[warn] ==== local: tried
[warn]   /Users/jkutner/.ivy2/local/com.typesafe.sbtrc/actor-client-2-11/0.3.1/ivys/ivy.xml
[warn] ==== public: tried
[warn]   https://repo1.maven.org/maven2/com/typesafe/sbtrc/actor-client-2-11/0.3.1/actor-client-2-11-0.3.1.pom
[info] Resolving jline#jline;2.12.1 ...
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	::          UNRESOLVED DEPENDENCIES         ::
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	:: com.typesafe.sbtrc#actor-client-2-11;0.3.1: not found
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 
[warn] 	Note: Unresolved dependencies path:
[warn] 		com.typesafe.sbtrc:actor-client-2-11:0.3.1 (/Users/jkutner/workspace/heroku/kissaten/activator-bug/build.sbt#L7-10)
[warn] 		  +- default:activator-bug_2.11:1.0
...
[error] (*:update) sbt.ResolveException: unresolved dependency: com.typesafe.sbtrc#actor-client-2-11;0.3.1: not found
[error] Total time: 1 s, completed Jul 15, 2015 9:52:31 AM
```

But with Activator, it is resolved:

```sh-session
$ activator update
...
[info] downloading https://repo.typesafe.com/typesafe/ivy-releases/com.typesafe.sbtrc/actor-client-2-11/0.3.1/jars/actor-client-2-11.jar ...
[info] 	[SUCCESSFUL ] com.typesafe.sbtrc#actor-client-2-11;0.3.1!actor-client-2-11.jar (1613ms)
...
[info] Done updating.
[success] Total time: 47 s, completed Jul 15, 2015 9:51:56 AM
```

Is difference is that Activator is using `ivyStylePatterns` for the `"https://repo.typesafe.com/typesafe/ivy-releases"` repo, while SBT is using Maven style.

```
resolvers += Resolver.url("Typesafe Ivy releases", url("https://repo.typesafe.com/typesafe/ivy-releases"))(Resolver.ivyStylePatterns)
```

Related issues:
https://github.com/playframework/playframework/issues/4839
https://github.com/typesafehub/activator/issues/979
https://github.com/sbt/sbt/issues/2054

**Comment by jkutner** [comment]
This manifests itself with the `sbt-fork-run-plugin`, which is added by Activator in `project/play-fork-run.sbt`:
https://github.com/typesafehub/activator/blob/7618e81feb658eb0904ae471d06c08edf269753f/ui/app/assets/services/sbt/configuration.js#L26

**Comment by corruptmemory** [comment]
The `sbt-fork-run-plugin` is not needed for anything other than the Activator UI.  With Activator 1.3.x all the "run"-like tasks are run in the background via sbt-server.  The plugin in question provides a "background run" that works like Play's developer mode: recompiling and reloading.

You do not need this plugin in your CI environment for anything.  I would suggest removing it and adding it to your `.gitignore`.

**Comment by jkutner** [comment]
@corruptmemory this is not my app. I work at Heroku, and I'm seeing lots of people encounter this error upon deployment. I'm trying to get it fixed for them. 

Regardless, there is bigger problem: Activator and SBT use different resolvers! I would think that's a big deal.

**Comment by corruptmemory** [comment]
Right, understood.

Again, one's first line of defense is to remove that plugin file.  We're planning to make that file unnecessary in the future.

Regarding the resolver, that too is going to be addressed - by removing the necessity to monkey with `~/.sbt/resolvers`.  A second thing to do is remove lines beginning `activator-*` from your `~/.sbt/repositories` file.

**Comment by jkutner** [comment]
@jsuereth how is this an enhancement? This literally breaks raw usage of SBT.

Instead of monkeying with `~/.sbt/repositories`, shouldn't the resolvers go in `project/play-fork-run.sbt`?

See also #1058

**Comment by hakanernam** [comment]
I strongly agree it is not an "enhancement". For some other reason I had to delete .ivy directory. After that I could not run 
```
sbt eclipse
```
failing with this exact error. Reading this thread,  I first  ran 
```
activator compile
```
which worked. And after that 

```
sbt eclipse
```
also worked.  Even if you call this not a bug, it definitely is extremely confusing/annoying/frustrating.

Also https://github.com/typesafehub/activator/issues/1058 suggests this happens on windows. 
But my system has:

MacOSX 10.11.2
JDK  "1.8.0_66"
sbt 0.13.9

**Comment by mslinn** [comment]
I'm hitting this same issue when trying to use SBT with a project on Windows that works fine on Linux

**Comment by antonin42** [comment]
+1 this bug just made me loose 30 min. This bug is open on different repos (plugin, sbt, activator) but the ball is thrown from one team to the other.
